File: Python/quicksort/ChatGPT/ChatGPT.py
from random import randint
def quicksort(arr):
    """Sorts an array using the Quicksort algorithm."""
    # Base case: arrays with fewer than two elements are already sorted
    if len(arr) < 2:
        return arr
    
    # Choose the pivot (here we choose the last element)
    pivot = arr[-1]
    
    # Partitioning step
    lesser = [x for x in arr[:-1] if x <= pivot]  # Elements less than or equal to pivot
    greater = [x for x in arr[:-1] if x > pivot]  # Elements greater than pivot
    
    # Recursively apply quicksort to subarrays and combine
    return quicksort(lesser) + [pivot] + quicksort(greater)

def main():
    # Generate 1000 random values
    random_values = [randint(1, 10000) for _ in range(1000)]

    # Sort the generated random values using quicksort
    sorted_values = quicksort(random_values)

if __name__ == "__main__":
    main()

File: C/quicksort/chatGPT/chatGPT.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void quicksort(int arr[], int low, int high) {
    if (low < high) {
        int pivot = partition(arr, low, high);
        quicksort(arr, low, pivot - 1);
        quicksort(arr, pivot + 1, high);
    }
}

int partition(int arr[], int low, int high) {
    int pivot = arr[high];
    int i = low - 1;
    for (int j = low; j < high; j++) {
        if (arr[j] < pivot) {
            i++;
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }
    int temp = arr[i + 1];
    arr[i + 1] = arr[high];
    arr[high] = temp;
    return i + 1;
}

int main() {
    int n = 1000;
    int arr[n];
    srand(time(NULL));
    
    // Generate 1000 random values
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % 10000;  // Random values from 0 to 9999
    }

    // Sort the array
    quicksort(arr, 0, n - 1);

    return 0;
}
